app/models/concerns/account_interactions.rb

```diff
@@ -117,9 +117,12 @@ def block!(other_account, uri: nil)
                        .find_or_create_by!(target_account: other_account)
   end
 
-  def mute!(other_account, notifications: nil)
+  def mute!(other_account, notifications: nil, duration: 0)
     notifications = true if notifications.nil?
-    mute = mute_relationships.create_with(hide_notifications: notifications).find_or_create_by!(target_account: other_account)
+    mute = mute_relationships.create_with(hide_notifications: notifications).find_or_initialize_by(target_account: other_account)
+    mute.expires_in = (duration.zero? ? nil : duration)
```

Comment:
I would remove the parenthesis here but otherwise it's fine.

Reply:
I removed the parentheses.